streams/transform-streams/cancel.any.js

```diff
@@ -0,0 +1,115 @@
+    cancel(reason) {
+      assert_equals(reason, originalReason, 'transformer.cancel() should be called with the passed reason');
+      throw thrownError;
+    }
+  });
+  const writer = ts.writable.getWriter();
+  const cancelPromise = ts.readable.cancel(originalReason);
+  await promise_rejects_exactly(t, thrownError, cancelPromise, 'readable.cancel() should reject with thrownError');
+  await promise_rejects_exactly(t, thrownError, writer.closed, 'writer.closed should reject with thrownError');
+}, 'cancelling the readable side should reject if transformer.cancel() throws');
+
+promise_test(async t => {
+  let aborted = undefined;
+  const ts = new TransformStream({
+    cancel(reason) {
+      aborted = reason;
+    },
+    flush: t.unreached_func('flush should not be called')
+  });
+  const res = await ts.writable.abort(thrownError);
+  assert_equals(res, undefined, 'writable.abort() should return undefined');
+  assert_equals(aborted, thrownError, 'transformer.abort() should be called with the passed reason');
+}, 'aborting the writable side should call transformer.abort()');
+
+promise_test(async t => {
+  const ts = new TransformStream({
+    cancel(reason) {
+      assert_equals(reason, originalReason, 'transformer.cancel() should be called with the passed reason');
+      throw thrownError;
+    },
+    flush: t.unreached_func('flush should not be called')
+  });
+  const reader = ts.readable.getReader();
+  const abortPromise = ts.writable.abort(originalReason);
+  await promise_rejects_exactly(t, thrownError, abortPromise, 'writable.abort() should reject with thrownError');
+  await promise_rejects_exactly(t, thrownError, reader.closed, 'reader.closed should reject with thrownError');
+}, 'aborting the writable side should reject if transformer.cancel() throws');
+
+promise_test(async t => {
+  const ts = new TransformStream({
+    async cancel(reason) {
+      assert_equals(reason, originalReason, 'transformer.cancel() should be called with the passed reason');
+      throw thrownError;
+    },
+    flush: t.unreached_func('flush should not be called')
+  });
+  const cancelPromise = ts.readable.cancel(originalReason);
+  const closePromise = ts.writable.close();
+  await Promise.all([
+    promise_rejects_exactly(t, thrownError, cancelPromise, 'cancelPromise should reject with thrownError'),
+    promise_rejects_exactly(t, thrownError, closePromise, 'closePromise should reject with thrownError'),
+  ]);
+}, 'closing the writable side should reject if a parallel transformer.cancel() throws');
+
+promise_test(async t => {
+  let controller;
+  const ts = new TransformStream({
+    start(c) {
+      controller = c;
+    },
+    async cancel(reason) {
+      assert_equals(reason, originalReason, 'transformer.cancel() should be called with the passed reason');
+      controller.error(thrownError);
+    },
+    flush: t.unreached_func('flush should not be called')
+  });
+  const cancelPromise = ts.readable.cancel(originalReason);
```

Comment:
While implementing in Firefox I found that this test fails. I am not 100% sure, but I think it's wrong. @lucacasonato @MattiasBuelens 

`ReadableStreamCancel` calls `ReadableStreamClose`, which does
> Step 2. Set stream.[[state]] to "closed".

The `controller.error` is basically ignored, because [`ReadableStreamDefaultControllerError`])(https://streams.spec.whatwg.org/#readable-stream-default-controller-error) returns early in
> Step 2. If stream.[[state]] is not "readable", return.

That means for the `TransformStreamDefaultSourceCancelAlgorithm` the `cancelPromise` is fullfilled and `writable.[[state]]` is `"closed"`, so "otherwise" is executed and the promise is resolved with undefined.